### ISSUE TITLE
Support for RUT over 100 million

### ISSUE THREAD

**Issue by noquierouser** [Laragear/Rut#25]
### Please check these requirements

- [X] This feature helps everyone using this package
- [X] It's feasible and maintainable
- [X] It's non breaking
- [ ] I issued a PR with the implementation (optional)

### Description

According to [the guidelines of the Ministry of Education](https://www.mineduc.cl/mineduc-anuncia-la-creacion-del-identificador-provisorio-escolar-ipe-estudiantes-migrantes/), since 2016, migrant students awaiting regularization are assigned a RUT over 100 million. This RUT is temporary and allows them access to services provided by the ministry, known as the Temporary School Identifier (Identificador Provisorio Escolar, IPE).

A few months ago, I had to implement a system that registers children. There were a couple of cases with RUTs over 100 million, which the system considered invalid.

This would entail:
- Raising the maximum RUTs that can be validated to above 100 million. (To 150 million?).
- Methods that allow determining if a RUT is temporary or definitive (`isTemporary`, `isDefinitive`).

### Code sample

```php
$rut = Rut::parse('100.765.432-5');

$rut->isValid(); //returns true

$rut->isPerson(); //returns true
$rut->isCompany(); //returns false

$rut->isTemporary(); //returns true
$rut->isDefinitive(); //returns false
```

**Comment by DarkGhostHunter** [comment]
I will work on this once I have time early next month, unless there is a PR before it put my hands on this enhancement. 

About the generator, it would also be to be updated so it's able to generate over 100M numbers. 

I think that raising the validity to 200M would be enough, since I don't know exactly how is the pace of RUT assignations.

**Comment by noquierouser** [comment]
Based on what I've been able to find, the occupancy rate of RUTs is quite low. I couldn't locate official documents, only these journalistic notes ([La Nación](https://www.lanacion.cl/registro-civil-refuta-supuesta-escasez-de-numeros-rut/), [Publimetro](https://www.publimetro.cl/cl/noticias/2018/04/25/podria-colapsar-sistema-actual-registro-civil-otorga-mil-numeros-diarios-run-extranjeros.html)) that report on both the usage rate and the RUT assignment ranges.

I spent hours searching for an official document that indicates the assignment ranges of the RUTs, but I could only find some references to Resolutions from the SII on the matter:
- [Resolución 5412/2000](https://www.sii.cl/pagina/actualizada/noticias/1212not10001.htm)
- [Pregunta ID 001.105.4033.008](https://www.sii.cl/preguntas_frecuentes/rut_inicio_actividades/001_105_4033.htm)
- [Pregunta ID 001.105.3235.007](https://www.sii.cl/preguntas_frecuentes/rut_inicio_actividades/001_105_3235.htm)

**Comment by DarkGhostHunter** [comment]
From what you shared:

- RUT for a Person is below 60M
- RUT for Companies is equal or above 60M
- Over 100M, there are Temporal.

This package should follow suit.

**Comment by DarkGhostHunter** [comment]
Update! Version 2.0 coming out. Breaking change, most notably are the RUT types boundaries:

- Person RUT from 100.000 to 45.999.999
- Foreign Investor Person RUT from 46.000.000 to 46.999.999
- Foreign Investor Company RUT from 47.000.000 to 47.999.999
- Contingency RUT from 48.000.000 to 59.999.999
- Company RUT from 60.000.000 to 99.999.999
- Temporal RUT from 100.000.000 to 199.999.999

In other words, a RUT 54.XXX.XXX no longer will be validated as "person", but for "contingency", since these are reserved for special reasons.

**Comment by DarkGhostHunter** [comment]
Fixed on v2.0.0
